### PR TITLE
Bash Scripts to Simplify Installs

### PR DESCRIPTION
**Pull request by stirringhalo**
Looking through the issues, I saw a few mentions for a desire to move away from docker due to newbie confusion (and perhaps difficulty in modifying and actually tracking what dependencies are needed for production installs). It also doesn't make much sense having a postgres database in a docker container as it isn't something ephemeral and just obfuscates where the actual DB is. Plus as a GIS guy, I can vouch that a lot of other GIS guys don't have the wider Comp Sci experience!

Tasks:
- [ ] postgis
- [ ] import-external
- [ ] import-osm
- [ ] import-sql
- [ ] generate-jobs
- [ ] export
- [ ] RabbitMQ
- [ ] merge-jobs
- [ ] changed-tiles
- [ ] create-extracts
- [ ] cleanup scripts for all

I'm going on the assumption of Ubuntu 14.04. I will be repeatedly testing all the scripts to make sure I haven't missed anything.

If you agree, don't merge this yet, I will be working through each of the scripts and moving them from docker to bash. I will keep containers like the export-workers as is as that makes sense as it can be rather ephemeral.

I'm not sure what you all prefer to make contributions and development with. Hopefully we switch or the changes are few enough that they can be separated into the standalone bash scripts. I don't mind track the changes and keeping up to date.

### CODE REVIEW

**Comment by lukasmartinelli** [comment]
First again thanks for digging into it and putting so much thought into it.
We really appreciate the effort. I usually want to merge PR's not decline them.

And excuse the following personal opinions, don't take it personally.

**tldr: This is the wrong approach**. 

> Looking through the issues, I saw a few mentions for a desire to move away from docker due to newbie confusion (and perhaps difficulty in modifying and actually tracking what dependencies are needed for production installs).

This was always targeted at the tileserver or that people who are serving tiles don't have to use  Docker (since they don't know it).

But for the entire workflow **we want to stay with Docker**. And I know I am speaking on behalf of @manuelroth and @klokan as well.

For the entire workflow which consists of so many dependencies and programs Docker is the best thing that happened. Without Docker the installation would be a nightmare, it provides us the utilities to completly isolate the dependencies and makes it much easier to execute compared to Kartotherian or Mapzen Vector tiles where you need to write manual scripts.

 And the entire cloud computing space has moved so quickly in the in the last two years towards Docker that this is only confirmed.

> It also doesn't make much sense having a postgres database in a docker container as it isn't something ephemeral and just obfuscates where the actual DB is.

Running the database in a Docker container is absolutely fine and it scales incredibly well and makes it very easy to scale to many hosts. So many cloud providers do this and at the last companies I worked with we did exactly the same at big scale. And the container might be ephemeral but the volumes mounted to the data directory is not.

Perhaps some people want to reuse an existing PostGIS DB and I understand that and this is still possible with the workflow by replacing the OSM connection env vars.

> I'm not sure what you all prefer to make contributions and development with. Hopefully we switch or the changes are few enough that they can be separated into the standalone bash scripts. I don't mind track the changes and keeping up to date.

I think this is the wrong approach to continue to work on this as part of osm2vectortiles.
You are however completely free to do this in a separate repo.

Again we appreciate the effort and thoughts here but it is the wrong direction for the osm2vectortiles flow since we are really happy in using Docker in the workflow in the last half year and were always confirmed that it is the right choice by our deployment experiences.

**Comment by ImreSamu** [comment]
>But for the entire workflow we want to stay with Docker

+1  

> Running the database in a Docker container is absolutely fine

in the [PGConf EU 2016] (https://wiki.postgresql.org/wiki/PostgreSQL_Conference_Europe_Talks_2015) at least 2 presentation were about Database Containerization:  
* Mladen Marinović :  ["Dockerizing a Larger PostgreSQL Installation: What could possibly go wrong?"](https://bitbucket.org/marin/pgconfeu2015/raw/883632334c68663e10ab9637003f32bc2fadb279/Dockerizing%20a%20Largrer%20PostgreSQL%20Installation.pdf)
* Honza Horak:  ["Database containers in enterprise world"](https://github.com/hhorak/presentations/blob/master/2015/2015-database-containers-in-enterprise-world.pdf)


And Docker is a good way to improve reproducibility in software engineering research:
 * https://scholar.google.com/scholar?q=Docker+containers+repeatability+reproducibility+science

**Comment by stirringhalo** [comment]
Ha! Fair enough :)

And I understand from the perspective of the tileserver. I do see some of the advantages of docker but also some painful points, especially w.r.t. the DB.
Keep in mind I'm coming from a mostly GIS perspective, with some CS but not necessarily heavily into development/deployment.

What are your thoughts on using "VOLUME /var/lib/postgresql" (to have it on direct filesystem) or have some option to connect into an existing backend DB? I realized backups can still be done easily with pg_dump as it is now (thanks!) but in cases of mixed/specialty storage, you might want to put the DB elsewhere other than the container.

**Comment by lukasmartinelli** [comment]
> What are your thoughts on using "VOLUME /var/lib/postgresql" (to have it on direct filesystem) or have some option to connect into an existing backend DB? I realized backups can still be done easily with pg_dump as it is now (thanks!) but in cases of mixed/specialty storage, you might want to put the DB elsewhere other than the container.

You're right.
In production I mount `/var/lib/postgresql` to an actual filesystem and separate partition.
To do that modify the docker-compose file at the `pgdata` data container.

This way I can do snapshot of disks etc. Working only with data containers can be cumbersome sometime.
